### PR TITLE
Show outdated warning - Fixes #719

### LINKED ISSUE #719: Mark old-release Ore guidelines as outdated
Could the older guidelines (I believe just 6.0.0 and 5.1.0 at present) be updated to indicate they are not the present version of the docs, and link to the latest (stable?)? Some sort of message at the top of the page, I would propose.

### PR DESCRIPTION
**Pull request by ST-DDT**
Adds a warning to the index page and the ore submission guidelines that they are referring to an old version of the guidelines.

Fixes https://github.com/SpongePowered/SpongeDocs/issues/719.

If this gets merged it has to be cherry-picked for `v5.1.0` as well.

### CODE REVIEW

File: source/index.rst [hunk 1]
```diff
@@ -2,6 +2,12 @@
 Sponge Documentation
 ====================
 
+.. warning::
+
+    This documentation refers to outdated API version and is no longer actively maintained. While the code examples
```
Comment (mbax) [inline source/index.rst]: "refers to *an* outdated API version" ?

Comment (mbax) [reply source/index.rst]: And on the other one too. Though it doesn't really refer to any API version there...

**Review by mbax — CHANGES_REQUESTED** [review]

File: source/index.rst (hunk 1 above)
Comment (ST-DDT) [reply source/index.rst]: Fixed. I don't know how to express the "its outdated" part otherwise.

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/assets.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/faq.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/future.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/glossary.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/introduction.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/license.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/posting.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/rules.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/staff.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/about/structure.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/contributing/guidelines.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/contributing/implementation/codestyle.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/contributing/implementation/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/contributing/implementation/pr.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/contributing/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/contributing/spongedocs.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/index.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/16e1bef/ore/guidelines.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/720#issuecomment-405373938), not to the direct url to the preview._

**Review by mbax — APPROVED** [review]

File: source/index.rst [hunk 2]
```diff
@@ -2,6 +2,12 @@
 Sponge Documentation
 ====================
 
+.. warning::
+
+    This documentation refers to an outdated API version and is no longer actively maintained. While the code examples
```
Comment (Inscrutable) [inline source/index.rst]: API -> SpongeAPI

File: source/index.rst [hunk 3]
```diff
@@ -2,6 +2,12 @@
...
+    still work for that API version, the policies, guidelines, and some links might have changed. Please refer to the
```
Comment (Inscrutable) [inline source/index.rst]: might -> may

File: source/ore/guidelines.rst [hunk 4]
```diff
@@ -2,6 +2,11 @@
 Ore Plugin Submission Guidelines
 ================================
 
+.. warning::
+
+    This document refers to an outdated API version and is no longer actively maintained. The policies, guidelines, and
```
Comment (Inscrutable) [inline source/ore/guidelines.rst]: API -> SpongeAPI

**Review by Inscrutable — COMMENTED** [review]
Changes not mandatory but desirable. 
Should we also have similar warnings in the regular Rules, Privacy Policy and TOS pages that refers to the latest docs as definitive?

**Comment by Inscrutable** [comment]
Wow. That's a lot of warnings. 
I'm starting to think having a general banner-type warning for all old Documentation pages might be better, although that might not be possible either. @Minecrell Do you think there's a way we can insert some general warning about outdated documentation when people view old versions of SpongeDocs? 
If that's not possible, then I guess this will PR do as is.

**Comment by Inscrutable** [comment]
Thinking more about this, if we did have that kind of "meta-warning" on outdated Docs, it would have to be part of the Docs (and not the theme) to get translated. Which probably rules that idea out. 
I don't know if it's worth the effort to de-clutter these pages from warnings. We could just put one big warning on the Main Index, but there's no guarantee it would even be seen. As long as warnings end up on critical pages (Rules, TOS, Privacy) to avoid issues it will do.
Maybe in the long term we can shoe-horn a statement "that the latest version should always be consulted" into the TOS. The current TOS says
> This Agreement constitutes the entire agreement between SpongePowered and you concerning the subject matter hereof, and they may only be modified by a written amendment signed by an authorized executive of SpongePowered, or by the posting by SpongePowered of a revised version. ``

So we could specify exactly where to find the current version, but it does imply latest is the active one.

Which is a long winded way of saying, yeah I think I should merge this as-is, unless anyone else wants to jump on the bonfire.

**Review by 12awsomeman34 — APPROVED** [review]

**Comment by Inscrutable** [comment]
All ready for Cherry-picking, I guess.

**Comment by ST-DDT** [comment]
Cherry picked to `v5.1.0`.

**Pull request merged**